perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -1198,6 +1174,7 @@ public static synchronized void refreshAuthz(PerunSession sess) throws InternalE
 		log.debug("Refreshing authz roles for session {}.", sess);
 
 		sess.getPerunPrincipal().setRoles(AuthzResolverBlImpl.getRoles(sess));
```

Comment:
This should be:
```java
// Prepare first users rights on all subgroups of groups where user is GroupAdmin and add them to AuthzRoles of the user
AuthzRoles authzRoles = addAllSubgroupsToAuthzRoles(sess, authzResolverImpl.getRoles(sess.getPerunPrincipal().getUser()));
```
like in ``init(sess)`` method.

Reply:
It is, the AuthzResolverBlImpl.getRoles() method calls addAllSubgroupsToAuthzRoles()

Reply:
Sorry, I overlooked you call bl method and not impl.